### PR TITLE
Update branch default to main branch

### PR DESCRIPTION
**Pull request by Mehrnaz-Charkhchi**
### Description
Replace dev branch with main in Tools API repo

### Related JIRA Issue(s)
[ENSWBSITES-2782](https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2782)

### Review App URL(s) 
http://__BRANCH_NAME_HERE__.review.ensembl.org

### Checklist

- [ ] Black formatting
- [ ] Tests

### CODE REVIEW

**Review by kamaldodiya — CHANGES_REQUESTED** [review]
Shouldn't this PR be against main ?

**Comment by Mehrnaz-Charkhchi** [comment]
> Shouldn't this PR be against main ?

@kamaldodiya I'm resolving all the conflicts here so yes it can be against main, I want to make sure dev branch isn't affecting deployments anymore.

**Review by kamaldodiya — COMMENTED** [review]
I think pipeline wise it looks good to me I checked it by simulating the pipeline in the gitlab and it is as expected. Ultimately we won't be able to ascertain until this gets merged to main and we have few pipelines run. 
Just one thing I would like to highlight is that readability ( understanding ) pipeline is bit difficult.
it would be good to have consistency across deploy job
for example .
https://github.com/Ensembl/ensembl-web-tools-api/blob/branch_name_update/.gitlab-ci.yml#L179-L180

the job extends from deploy-main and explicit about stage
where as it changes for deploy production job 

https://github.com/Ensembl/ensembl-web-tools-api/blob/branch_name_update/.gitlab-ci.yml#L191-L193

 where the stage is inferred from the reply-main template.

**Comment by kamaldodiya** [comment]
Also https://github.com/Ensembl/ensembl-web-tools-api/blob/branch_name_update/.gitlab-ci.yml#L183-L184
Here branch is already inferred in the deploy-main template

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -173,29 +166,22 @@ Stop_review:
...
 Staging:
   extends:
-    - .deploy-dev
+    - .deploy-main
+  stage: deploy-preview
   environment:
     name: staging
   rules:
-    - if: $CI_COMMIT_BRANCH == "dev"
+    - if: $CI_COMMIT_BRANCH == "main"
```
Comment (kamaldodiya) [inline .gitlab-ci.yml]: The main in already inferred via .deploy-main template. I think this is redundant.

**Review by kamaldodiya — CHANGES_REQUESTED** [review]

**Comment by veidenberg** [comment]
@Mehrnaz-Charkhchi this PR includes commits and changes that have already been merged. Can you rebase `branch_name_update` to the last commit in `dev` branch?

**Comment by Mehrnaz-Charkhchi** [comment]
> @Mehrnaz-Charkhchi this PR includes commits and changes that have already been merged. Can you rebase `branch_name_update` to the last commit in `dev` branch?

@veidenberg I did explain during the code review it's ready to be merged to main I'm merging it to dev to check the pipeline one last time.

**Review by kamaldodiya — APPROVED** [review]

**Pull request merged**